### PR TITLE
Adds the Nuclear training bomb

### PR DESCRIPTION
**Pull request by datlo**
## What Does This PR Do

This PR adds the training nuclear bomb, paper guides, aswell two training authentification disks, a toolbox, a multitool, and a welding helmet to the brig on all maps. This is either added to the firing range or to a new Defusal Workshop room, depending on map.

The training nuke functions the same as the regular nuke, letting you practice with arming, disarming, or unanchoring the nuke. 'Detonating' the training nuke resets it. You can also manually reset it by alt-clicking it.

Wires have been reworked, defusing nuke now requires you cut the control panel wire, then pulse the disarm wire.
NT nukes now have labelled wires, syndicate nukes are still hidden. Keep in mind you have to use the NAD to access the nuke's wires.

Cleaned up some code, added span to messages that missed it.

## Why It's Good For The Game

Letting people train with a mechanic crucial to the nuclear gamemode without having to boot up their own server is good

Defusing the nuke using wires used to be basically just 'pick wire at random pray for best'. It is now a little more involved, the game giving you hints on how to proceed aswell.
## Images of changes

Training nuke sprite by McRamon, training NAD by ATPEngineer

Box : 
image

Delta : 
image

Meta : 
image

Ceres : 
image

Guides : 

nukeguides

## Testing

Tested every mechanic repeatedly on both the new training and the standard nukes. Launched all the maps to check for mapping errors.

## Changelog
:cl:
add: Added the nuclear training bomb to the security firing range on all maps. Training nuke sprite by McRamon, training NAD sprites by ATP Engineer.
/:cl:

### CODE REVIEW

**Comment by Fluffyzack21** [comment]
Should this also be on meta, delta, and farrgus?

**Comment by datlo** [comment]
They will be, it's on the todo list

**Comment by 1080pCat** [comment]
Recon you could add a cmag effect to make it trigger the delta announcement?

**Comment by datlo** [comment]
I'm considering emag/cmag effect, not sure about announcement tho

File: code/modules/paperwork/paper.dm [hunk 1]
```diff
@@ -691,6 +691,42 @@
 	header = "<p><img style='display: block; margin-left: auto; margin-right: auto;' src='ntlogo.png' width='220' height='135' /></p><hr />"
 	info =  ""
 
+/obj/item/paper/nuclear_guide_operating
+	name = "nuclear guide : 'Operating the Nuclear Device'"
+	info = "<b>ARMING THE NUCLEAR DEVICE</b><hr /> \
+		1. Insert the NAD <br> \
```
Comment (Henri215) [inline code/modules/paperwork/paper.dm:697]:
```suggestion
		1. Insert the NAD. <br> \
```

**Comment by Henri215** [comment]
image

What has been modified on those?

**Comment by datlo** [comment]
> image
> 
> What has been modified on those?

Good question, nothing should have changed. I'll see what I can do to fix that

File: code/modules/research/designs/misc_designs.dm [hunk 2]
```diff
@@ -51,6 +51,16 @@
 	build_path = /obj/item/disk/tech_disk
 	category = list("Miscellaneous")
 
+/datum/design/tech_disk
```
Comment (Henri215) [inline code/modules/research/designs/misc_designs.dm:54]:
```suggestion
/datum/design/training_disk
```

**Comment by datlo** [comment]
Fixed the issue with the NAD sprite changing, there was a rogue pixel.

I've downloaded and used the nuclear bomb icon file from the main repo but i'm still getting false positive on nuclearbombc. as far as I can tell this is a weird icondiff issue, can't seem to fix it. Nothing was changed on that sprite, I'm pretty sure.

**Comment by FunnyMan3595** [comment]
We like the idea of this, but the current sprite is giving us the wrong vibe, Please speak with our sprite team in Discord to figure out how to adjust it.

**Comment by datlo** [comment]
I'll deconflict this once I have a new sprite for the bomb. Id like something closer to crash test dummies.

File: code/game/gamemodes/nuclear/nuclearbomb.dm [hunk 3]
```diff
@@ -68,9 +70,16 @@ GLOBAL_VAR(bomb_set)
...
+		wires.labelled = FALSE
+		ADD_TRAIT(src, TRAIT_OBSCURED_WIRES, ROUNDSTART_TRAIT)
 	previous_level = SSsecurity_level.get_current_level_as_text()
-	GLOB.poi_list |= src
+	if(!training)
+		GLOB.poi_list |= src
+		if(is_syndicate)
+			GLOB.syndi_nuke_list |= src
+		else
+			GLOB.nt_nuke_list |= src
```
Comment (Furrior) [inline code/game/gamemodes/nuclear/nuclearbomb.dm:82]: This should be done oop way, so redefining stuff in subclasses. Eg adding a proc add_to_nuke_list which ll add to nt list and redefine it in syndie bomb. Same with labeling and training

**Review by Furrior — CHANGES_REQUESTED** [review]

File: code/game/gamemodes/nuclear/nuclearbomb.dm (hunk 3 above)
Comment (datlo) [reply code/game/gamemodes/nuclear/nuclearbomb.dm]: Adressed, moved stuff to initialize subprocs

**Comment by datlo** [comment]
Updated sprite, adressed review and fix conflict

**Review by Furrior — APPROVED** [review]
I would do it a little differently, but fine. Looks henry to me

**Review by FunnyMan3595 — CHANGES_REQUESTED** [review]

File: code/datums/wires/nuclearbomb_wires.dm [hunk 4]
```diff
@@ -18,39 +19,63 @@
...
+				N.audible_message("<span class='boldnotice'>The disarm controls flash with an error. You need to disable the control panel first!</span>", hearing_distance = 1)
+			else if(!is_cut(WIRE_NUKE_CONTROL))
+				N.audible_message("<span class='notice'>The disarm controls flicker.</span>", hearing_distance = 1)
+
+		if(WIRE_NUKE_LIGHT)
+			N.audible_message("<span class='notice'>The lights on [N] flicker.</span>", hearing_distance = 1)
+			flick(N.sprite_prefix + "nuclearbombc", N)
+
+		if(WIRE_NUKE_CONTROL)
+			N.audible_message("<span class='notice'>The control panel flicker.</span>", hearing_distance = 1)
```
[the remark this answers is not in the document]
Comment (DGamerL) [reply code/datums/wires/nuclearbomb_wires.dm:62]: Maybe
```suggestion
			N.audible_message("<span class='notice'>[N]'s control panel flickers.</span>", hearing_distance = 1)
```
?

**Comment by Burzah** [comment]
Please handle any merge conflicts and address any outstanding reviews when able.

File: code/_globalvars/lists/objects.dm [hunk 5]
```diff
@@ -35,6 +35,9 @@ GLOBAL_LIST_EMPTY(beacons)
 GLOBAL_LIST_EMPTY(shuttle_caller_list)  		//list of all communication consoles, comms consoles circuit and AIs, for automatic shuttle calls when there are none.
 GLOBAL_LIST_EMPTY(tracked_implants)			//list of all current implants that are tracked to work out what sort of trek everyone is on. Sadly not on lavaworld not implemented...
 GLOBAL_LIST_EMPTY(pinpointer_list)			//list of all pinpointers. Used to change stuff they are pointing to all at once.
+GLOBAL_LIST_EMPTY(nuke_list)				//list of (real) nukes
+GLOBAL_LIST_EMPTY(syndi_nuke_list)			//list of syndicate nukes
+GLOBAL_LIST_EMPTY(nad_list)					//list of (real) nuclear authentification disks
```
Comment (Henri215) [inline code/_globalvars/lists/objects.dm:40]:
```suggestion
GLOBAL_LIST_EMPTY(nad_list)					//list of (real) nuclear authentication disks
```

File: code/game/gamemodes/nuclear/nuclearbomb.dm [hunk 6]
```diff
@@ -710,6 +754,54 @@ GLOBAL_VAR(bomb_set)
...
+	training_reset()
+
+/obj/machinery/nuclearbomb/training/proc/training_reset()
+	if(auth)
+		auth.forceMove(get_turf(src))
+	new /obj/machinery/nuclearbomb/training(get_turf(src))
+	qdel(src)
+
+/obj/item/disk/nuclear/training
+	name = "training authentification disk"
```
Comment (Henri215) [inline code/game/gamemodes/nuclear/nuclearbomb.dm:798]:
```suggestion
	name = "training authentication disk"
```

**Comment by datlo** [comment]
I can't atm handle this PR, if anyone take it over and make it, it'd be appreciated.

**Pull request closed without merging**